src/tools/rust-analyzer/crates/proc-macro-srv/src/abis/abi_1_58/proc_macro/bridge/closure.rs

```diff
@@ -11,7 +11,7 @@ struct Env;
 impl<'a, A, R, F: FnMut(A) -> R> From<&'a mut F> for Closure<'a, A, R> {
     fn from(f: &'a mut F) -> Self {
         unsafe extern "C" fn call<A, R, F: FnMut(A) -> R>(env: &mut Env, arg: A) -> R {
-            (*(env as *mut _ as *mut F))(arg)
+            (&mut *(env as *mut _ as *mut F))(arg)
```

Comment:
As a side-note: this is UB!

`&mut Env` (where `Env` is a ZST) is used as an erased type and casted back to `F`. According to my understanding of Stacked Borrows(TM) this is UB, because pointers derived from `&mut Env` only have provenance over 0 bytes.

See also this comment: https://github.com/rust-lang/rust/pull/95308#discussion_r835372839

Reply:
Yeah SB doesn't like this, though I do hope that future aliasing models will accept such code.